### PR TITLE
🎉 New Source: ElasticSearch

### PR DESCRIPTION
**Pull request by olivermeyer**
## What
In this PR I create a new ElasticSearch source.

## Pre-merge Checklist
<p>
   
#### Community member or Airbyter
   
- [x] **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- [x] Secrets in the connector's spec are annotated with `airbyte_secret` 
- [x] Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- [ ] Code reviews completed
- [ ] Documentation updated 
    - [x] Connector's `README.md`
    - [ ] Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - [x] `docs/SUMMARY.md`
    - [x] `docs/integrations/<source or destination>/<name>.md` including changelog. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - [x] `docs/integrations/README.md`
    - [ ] `airbyte-integrations/builds.md`
- [x] PR name follows [PR naming conventions](https://docs.airbyte.io/contributing-to-airbyte/updating-documentation#issues-and-pull-requests)
- [ ] Connector added to connector index like described [here](https://docs.airbyte.io/connector-development#publishing-a-connector)
   
#### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items. 
   
- [ ] Create a non-forked branch based on this PR and test the below items on it
- [ ] Build is successful
- [ ] Credentials added to Github CI. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci). 
- [ ] [`/test connector=connectors/<name>` command](https://docs.airbyte.io/connector-development#updating-an-existing-connector) is passing. 
- [ ] New Connector version released on Dockerhub by running the `/publish` command described [here](https://docs.airbyte.io/connector-development#updating-an-existing-connector)
   
</p>

### CODE REVIEW

**Comment by marcosmarxm** [comment]
Hi @olivermeyer thanks for this contribution! Please sign the CLA and let me know when it's ready to review (change from draft to ready)

**Comment by olivermeyer** [comment]
Acceptance tests results:
```
$ python -m pytest integration_tests -p integration_tests.acceptance

Test session starts (platform: darwin, Python 3.7.9, pytest 6.1.2, pytest-sugar 0.9.4)
cachedir: .pytest_cache
rootdir: /Users/Omeyer/code/perso/airbyte, configfile: pytest.ini
plugins: cov-2.12.1, sugar-0.9.4, mock-3.6.1, timeout-1.4.2
collecting ... {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \nspec\ninput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_match_expected_inputs0_0/run_1/input\noutput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_match_expected_inputs0_0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_match_expected[inputs0] ✓                                                          8% ▉
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_required[inputs0] ✓                                                               17% █▋
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_optional[inputs0] ✓                                                               25% ██▌
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_has_secret[inputs0] ✓                                                             33% ███▍
 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_secret_never_in_the_output[inputs0] ✓                                             42% ████▎     {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \nspec\ninput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_oauth_flow_parameters_inp0/run_1/input\noutput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_oauth_flow_parameters_inp0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestSpec.test_oauth_flow_parameters[inputs0] ✓                                                  50% █████     {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \ncheck --config tap_config.json\ninput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_check_inputs0_0/run_1/input\noutput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_check_inputs0_0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestConnection.test_check[inputs0] ✓                                                            58% █████▉    {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \ncheck --config tap_config.json\ninput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_check_inputs1_0/run_1/input\noutput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_check_inputs1_0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestConnection.test_check[inputs1] ✓                                                            67% ██████▋   {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \ndiscover --config tap_config.json\ninput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_discover_inputs0_0/run_1/input\noutput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_discover_inputs0_0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestDiscovery.test_discover[inputs0] ✓                                                          75% ███████▌  {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \ndiscover --config tap_config.json\ninput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_defined_cursors_exist_in_0/run_1/input\noutput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_defined_cursors_exist_in_0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestDiscovery.test_defined_cursors_exist_in_schema[inputs0] ✓                                   83% ████████▍ {"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \nread --config tap_config.json --catalog catalog.json\ninput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_read_inputs0_0/run_1/input\noutput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_read_inputs0_0/run_1/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_core.py::TestBasicRead.test_read[inputs0] ✓                                                              92% █████████▎{"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \nread --config tap_config.json --catalog catalog.json\ninput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_sequential_reads_inputs0_0/run_1/input\noutput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_sequential_reads_inputs0_0/run_1/output"}}
{"type": "LOG", "log": {"level": "INFO", "message": "Docker run: \nread --config tap_config.json --catalog catalog.json\ninput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_sequential_reads_inputs0_0/run_2/input\noutput: /private/var/folders/__/sklc3lw52dndlzjgjkflkd8w0000gq/T/pytest-of-Omeyer/pytest-0/test_sequential_reads_inputs0_0/run_2/output"}}

 airbyte-integrations/bases/source-acceptance-test/source_acceptance_test/tests/test_full_refresh.py::TestFullRefresh.test_sequential_reads[inputs0] ✓                                       100% ██████████
========================================================================================= short test summary info ==========================================================================================
SKIPPED [1] ../../bases/source-acceptance-test/source_acceptance_test/plugin.py:56: Skipping TestIncremental.test_two_sequential_reads because not found in the config

Results (18.09s):
      12 passed
```

Unit tests results:
```
$ python -m pytest unit_tests

Test session starts (platform: darwin, Python 3.7.9, pytest 6.1.2, pytest-sugar 0.9.4)
cachedir: .pytest_cache
rootdir: /Users/Omeyer/code/perso/airbyte, configfile: pytest.ini
plugins: cov-2.12.1, sugar-0.9.4, mock-3.6.1, timeout-1.4.2
collecting ... {"type": "LOG", "log": {"level": "INFO", "message": "Getting indices from ElasticSearch"}}

 airbyte-integrations/connectors/source-elasticsearch/unit_tests/unit_test.py::test_get_indices ✓                                                                                             33% ███▍
 airbyte-integrations/connectors/source-elasticsearch/unit_tests/unit_test.py::test_get_index_json_properties_happy_path ✓                                                                    67% ██████▋
 airbyte-integrations/connectors/source-elasticsearch/unit_tests/unit_test.py::test_get_index_json_properties_unsupported_tyoe ✓                                                             100% ██████████

Results (0.49s):
       3 passed
```

**Comment by marcosmarxm** [comment]
thanks @olivermeyer will add this to our sprint to review.

File: airbyte-integrations/connectors/source-elasticsearch/source_elasticsearch/source.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+
+class UnsupportedDataTypeException(Exception):
+    pass
+
+
+class SourceElasticsearch(Source):
+    system_indices = [
+        ".kibana_1",
+        ".opendistro_security",
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-elasticsearch/source_elasticsearch/source.py:32]: I don't think this is the only system indices, should be better to have a specification option to read only the indices user want to.

File: airbyte-integrations/connectors/source-elasticsearch/source_elasticsearch/source.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+    ]
+
+    es_to_json_type_mapping = {
+        # TODO: support objects, arrays, etc
+        # ES data types: https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping-types.html
+        # JSON data types: http://json-schema.org/understanding-json-schema/reference/type.html
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-elasticsearch/source_elasticsearch/source.py:38]: Without objects and array merge this become very complicate @olivermeyer. They are very simple and native data types in ElasticSearch. Please make sure those types are supported.

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Overall looks great, but we need the support for array and object.

**Comment by marcosmarxm** [comment]
@olivermeyer let me know when you return to finish the change requested I made!

**Comment by marcosmarxm** [comment]
@olivermeyer  I'll close this due to inactivity. If you want to return this work you can reopen the PR.

**Pull request closed without merging**